packages/components/src/DateTimePickers/DateTime/Manager/Manager.component.js

```diff
@@ -29,38 +35,20 @@ function ContextualManager(props) {
 			props.onChange(event, { datetime, textInput, errors, errorMessage });
 		}
 	}
-	function onDateChange(event, { date, localDate, textInput: dateTextInput, errors = [] }) {
-		let datetime;
-		if (errors.length > 0) {
-			datetime = null;
-		} else {
-			datetime = dateAndTimeToDateTime(localDate, state.time, getDateOptions());
-		}
+	function onDateChange(event, payload) {
+		const newState = updatePartsOnDateChange(payload, state.time, getDateOptions());
 		const nextState = {
 			...state,
-			date: date || dateTextInput,
-			datetime,
-			textInput: dateAndTimeToStr(date || dateTextInput, state.time, getDateOptions()),
-			errors,
-			errorMessage: errors[0] ? errors[0].message : null,
+			...newState,
 		};
 		setState(nextState);
 		onChange(event, nextState);
 	}
-	function onTimeChange(event, { time, textInput: timeTextInput, errors = [] }) {
-		let datetime;
-		if (errors.length > 0) {
-			datetime = null;
-		} else {
-			datetime = dateAndTimeToDateTime(state.date, time, getDateOptions());
-		}
+	function onTimeChange(event, payload) {
+		const newState = updatePartsOnTimeChange(payload, state.date, getDateOptions());
 		const nextState = {
 			...state,
-			time: time || timeTextInput,
-			datetime,
-			textInput: dateAndTimeToStr(state.date, timeTextInput, getDateOptions()),
-			errors,
-			errorMessage: errors[0] ? errors[0].message : null,
+			...newState,
```

Comment:
same here, these logic are moved to `updatePartsOnTimeChange` method in datetime-extraction.